usr/sbin/regionsrv-enabler-azure

```diff
@@ -12,6 +12,34 @@
 # Lesser General Public License for more details.
 # You should have received a copy of the GNU Lesser General Public
 # License along with this library.
+
+"""
+Check the different states of the license type:
+    START   LICENSE CHANGED TO     CACHE   ACTION
```

Comment:
We'll need to clarify what `START` means. `BYOS-2` at instance start is only possible if a customer  passes user data and makes cloud-init run the registration process via the passed data. This process is probably rare but worth to be considered. In any event if the "register via extra setup" is considered as `BYOS-2` it should be in the comment.

Reply:
but this states can happen at any time, correct ? if customer removed the cache file, and run the AHB command then this `BYOS-2` would be possible

Reply:
Which is why `START` as a term needs to be defined. Since this is a cloud-instance it is arguably somewhat "natural" to associate the word "start" with "the start of an instance". As you point out if a user messes with the cache, which they shouldn't but can, and we need to handle that case, then a certain condition can be triggered. This is why I pointed out that we need to define what `START` means.

Reply:
I defined `START` [here](https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/432d94a48cad7aebd71382edf9db5b64cbfe8e2a/usr/sbin/regionsrv-enabler-azure#L41)
```
START represents the state of the license type before running AHB
```

Reply:
Missed that, so let's have the definitions above the table.

That said, AHB is the name for a feature in Azure, it cannot be run. Maybe we can define `START` as follows:

"The state of the instance when this code is executed. This code is executed by a timer"